src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -1501,309 +1440,74 @@ private String setFields(String url, List<String> fields) {
         JsonObject outerObj = parseJson(url, json).getAsJsonObject();
         JsonObject innerObj = outerObj.get(wrapperObjectName).getAsJsonObject();
         
-        return objectToPage(innerObj, clazzType);
+        return JSONUtils.objectToPage(innerObj, clazzType);
     }
 
     private <T> Page<T> getPage(String url, Map<String,String> params, Class<T> clazzType, int expectedCode) throws ApiException {
         url = setParams(url, params);
         String json = doGet(url, HTTP_STATUS_OK);
         JsonElement obj = parseJson(url, json);
         
-        return objectToPage(obj.getAsJsonObject(), clazzType);
+        return JSONUtils.objectToPage(obj.getAsJsonObject(), clazzType);
     }
     
-    private <T> Page<T> objectToPage(JsonObject obj, Class<T> clazzType) {
-        Set<Map.Entry<String, JsonElement>> set = obj.entrySet();
-
-        Page<T> p = new Page<T>();
-
-        for (Map.Entry<String, JsonElement> a : set) {
-            String key = a.getKey();
-            JsonElement val = a.getValue();
-
-            if (key.equals("page")) {
-                p.setPage(val.getAsInt());
-                continue;
-            }
-            if (key.equals("pages")) {
-                p.setPages(val.getAsInt());
-                continue;
-            }
-            if (key.equals("count")) {
-                p.setCount(val.getAsInt());
-                continue;
-            }
-            if (key.equals("items") || key.equalsIgnoreCase("results")) {
-                p.setItems(getPageItems(val, clazzType));
-            }
-        }
-        return p;
-    }
-
-    private <T> ArrayList<T> getPageItems(JsonElement elem, Class<T> clazzType) {
-        JsonArray ar = elem.getAsJsonArray();
-        ArrayList<T> col = new ArrayList<T>(ar.size());
-        for (JsonElement e : ar) {
-            T o = (T) Parser.getInstance().getObject(e, clazzType);
-            if (o != null) {
-                col.add(o);
-            }
-        }
-        return col;
-    }
-
     private <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
         HttpURLConnection conn = null;
         try {
-            conn = getConnection(apiKey, url, METHOD_POST, requestBody != null);
-
-            if (requestBody != null) {
-                conn.setDoOutput(true);
-                OutputStream output = null;
-                try {
-                    output = conn.getOutputStream();
-                    output.write(requestBody.getBytes("UTF-8"));
-                } finally {
-                    if (output != null) {
-                        try { output.close(); } catch (IOException ignored) {}
-                    }
-                }
-            }
-            conn.connect();
-            checkStatusCode(conn, expectedCode);
+            conn = HTTPConnectionUtils.getConnection(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody);
```

Comment:
If you were able to wrap `HttpURLConnection` in a wrapper that would implement `AutoCloseable`, you could then write
``` java
try(MyConnection connecton = HTTPConnectionUtils.getConnection(..)) {
   ...
}
```
and you wouldn't need `finally` at all. What do you think?